epic_manager/models.py

```
@@ -62,9 +62,29 @@ def from_json(cls, json_str: str) -> 'EpicPlan':
+        if 'epic' not in data:
+            raise KeyError("Missing 'epic' field in JSON response")
+        if 'issues' not in data:
+            raise KeyError("Missing 'issues' field in JSON response")
+        if 'parallelization' not in data:
+            raise KeyError("Missing 'parallelization' field in JSON response")
+
+        # Parse epic info with better error messages
+        try:
+            epic = EpicInfo(**data['epic'])
+        except TypeError as e:
+            raise KeyError(f"Invalid epic data: {e}. Epic data: {data['epic']}")
```

Ruff (0.13.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)